brew install git fail (snow leopard)

brew install git 
==> Downloading https://www.kernel.org/pub/software/scm/git/git-2.42.0.tar.xz
Already downloaded: /Users/palo/Library/Caches/Homebrew/git-2.42.0.tar.xz
==> Patching
patching file sha1dc/sha1.c
patching file config.mak.uname
patching file Makefile
Hunk #1 succeeded at 1348 (offset -3 lines).
patching file contrib/credential/osxkeychain/git-credential-osxkeychain.c
==> gmake install prefix=/usr/local/Cellar/git/2.42.0 sysconfdir=/usr/local/etc CC=gcc-4.2 CFLAGS=-std=gnu99 LDFLAGS=
Last 15 lines from /Users/palo/Library/Logs/Homebrew/git/01.gmake:
gcc-4.2 -o sha1dc/sha1.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  sha1dc/sha1.c
gcc-4.2 -o sha1dc/ubc_check.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  sha1dc/ubc_check.c
gcc-4.2 -o sha256/block/sha256.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  sha256/block/sha256.c
gcc-4.2 -o compat/precompose_utf8.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  compat/precompose_utf8.c
gcc-4.2 -o compat/fopen.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  compat/fopen.c
gcc-4.2 -o compat/memmem.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  compat/memmem.c
gcc-4.2 -o compat/qsort_s.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  compat/qsort_s.c
gcc-4.2 -o compat/regcomp_enhanced.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  compat/regcomp_enhanced.c
gcc-4.2 -o compat/stub/procinfo.o -c     -std=gnu99 -I. -DPRECOMPOSE_UNICODE -DPROTECT_HFS_DEFAULT=1 -I/usr/local/include -I/usr/local/opt/gettext/include -DGIT_HOST_CPU="\"i386\"" -I/usr/local/opt/curl/include -DUSE_CURL_FOR_IMAP_SEND -DNO_OPENSSL -DUSE_ST_TIMESPEC -DNO_GETTEXT -DSUPPORTS_SIMPLE_IPC -DSHA1_DC -DSHA1DC_NO_STANDARD_INCLUDES -DSHA1DC_INIT_SAFE_HASH_DEFAULT=0 -DSHA1DC_CUSTOM_INCLUDE_SHA1_C="\"git-compat-util.h\"" -DSHA1DC_CUSTOM_INCLUDE_UBC_CHECK_C="\"git-compat-util.h\"" -DSHA256_BLK  -DHAVE_DEV_TTY -DHAVE_BSD_SYSCTL -DHAVE_OPENSSL_CSPRNG -DHAVE_NS_GET_EXECUTABLE_PATH  -DAPPLE_COMMON_CRYPTO -DFREAD_READS_DIRECTORIES -DNO_MEMMEM -DUSE_ENHANCED_BASIC_REGULAR_EXPRESSIONS -DHAVE_FSMONITOR_DAEMON_BACKEND -DHAVE_FSMONITOR_OS_SETTINGS -DSHELL_PATH='"/bin/sh"'  compat/stub/procinfo.c
compat/regcomp_enhanced.c: In function ‘git_regcomp’:
compat/regcomp_enhanced.c:7: error: ‘REG_ENHANCED’ undeclared (first use in this function)
compat/regcomp_enhanced.c:7: error: (Each undeclared identifier is reported only once
compat/regcomp_enhanced.c:7: error: for each function it appears in.)
make: *** [compat/regcomp_enhanced.o] Error 1
make: *** Waiting for unfinished jobs....

READ THIS: https://github.com/mistydemeo/tigerbrew/wiki/troubleshooting

These open issues may also help:
git: "dyld: incompatible cpu-subtype" ruby(10.4/G3) https://github.com/mistydemeo/tigerbrew/issues/272
building ghc 7.10.2 (dependency for git-annex) fails on Mountain Lion https://github.com/mistydemeo/tigerbrew/issues/658

--- a/Library/Formula/git.rb
+++ b/Library/Formula/git.rb
@@ -212,7 +212,7 @@ __END__
  		OLD_ICONV = UnfortunatelyYes
  		NO_APPLE_COMMON_CRYPTO = YesPlease
  	endif
-+	ifeq ($(shell expr "$(uname_R)" : '[156789]\.'),2)
++	ifeq ($(shell test "`expr "$(uname_R)" : '\([0-9][0-9]*\)\.'`" -lt 11 && echo 1),1)
 +		NO_REGEX=YesPlease
 +	else
 +		USE_ENHANCED_BASIC_REGULAR_EXPRESSIONS = YesPlease

